src/ElasticSearchPackage.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+var _ = require('underscore'),
+    BaseResource = require('./BaseResource'),
+    AWS = require('aws-sdk'),
+    elasticsearch = new AWS.ES(),
+    POLL_DELAY = 5000;
+
+module.exports = BaseResource.extend({
+
+   doCreate: async function(props) {
+      const createPackageParams = {
+         PackageName: props.PackageName,
+         PackageType: props.PackageType,
+         PackageDescription: props.PackageDescription,
+         PackageSource: {
+            S3BucketName: props.PackageSource.S3BucketName,
+            S3Key: props.PackageSource.S3Key,
+         },
+      };
+
+      const createPackageResp = await elasticsearch.createPackage(createPackageParams).promise(),
+            packageID = createPackageResp.PackageDetails.PackageID;
+
+      console.log(
+         `Created package ${createPackageResp.PackageDetails.PackageName} (${packageID}),`
+         + ` current status: ${createPackageResp.PackageDetails.PackageStatus}`
+      );
+
+      await this._waitForPackageToBeAvailable(packageID);
+
+      await elasticsearch
+         .associatePackage({
+            DomainName: props.DomainName,
+            PackageID: packageID,
+         })
+         .promise();
+
+      const domainPackageDetails = await this._waitForPackageToBeAssociatedToDomain(packageID, props.DomainName);
+
+      return {
+         PhysicalResourceId: packageID,
+         ReferencePath: domainPackageDetails.ReferencePath,
+      };
+   },
+
+   doDelete: async function(resourceID, props) {
```

Comment:
Just as a note, this cleanup process does not work very well when the initial resource creation fails. A developer will usually have to go into the AES console and manually delete the association/package. For the first iteration of this resource, I'm hesitant to add the extra logic for auto-recovering from the various failure states. Initial investigations seem to indicate doing so will add a fair amount of complexity.